eZ/Publish/Core/Repository/URLAliasService.php

```diff
@@ -73,12 +74,17 @@ public function __construct(RepositoryInterface $repository, Handler $urlAliasHa
      * @param string $languageCode the languageCode for which this alias is valid
      * @param bool $alwaysAvailable
      *
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException if the user is not allowed to create url alias
      * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException if the path already exists for the given language
      *
      * @return \eZ\Publish\API\Repository\Values\Content\URLAlias
      */
     public function createUrlAlias(Location $location, $path, $languageCode, $forwarding = false, $alwaysAvailable = false)
     {
+        if ($this->repository->hasAccess('content', 'urltranslator') !== true) {
```

Comment:
This is deprecated, please use `$this->repository->getPermissionResolver()->hasAccess()`.

I'm afraid your mocks need to be changed as well in that case. For us integration tests are more important (the ones which @adamwojs mentioned).

So you could:
- align mocks existing prior to your PR changes so they don't crash,
- skip implementing new mocks if you don't want to spend extra time fixing them,
- implement integration testing.

Of course you can implement both mocks and integration tests if you have time :-)